src/niobot/client.py

```diff
@@ -813,11 +831,24 @@ async def send_message(
 
         if isinstance(room, nio.MatrixUser) or (isinstance(room, str) and room.startswith("@")):
             _user = room
+            room = None
             rooms = await self.get_dm_rooms(_user)
+            logging.debug( f"send_message get_dm_rooms returns rooms {nio.Api.to_json(rooms)}" )
+            
             if rooms:
-                room = rooms[0]
-            else:
-                room = await self.create_dm_room(_user)
+                for r_id in rooms:
+                    if r_id in self.rooms:
+                        room = r_id
+                        break
+                    else:
+                        logging.warning(f"room {r_id} not found in bot.rooms")
+            if not room:
+                logging.info("creating dm room for user {_user}")
+                response = await self.create_dm_room(_user)
+
+                room = self.rooms.get(response.room_id)
+                if not room:
+                    raise RuntimeError("DM room %r was created, but could not be found in the room list!" % room_id)
```

Comment:
Could you add documentation for the RuntimeError exception into the docstring? only ValueError and MessageException are documented at the moment